--- rpm-5.4.10/configure.ac~	2012-07-18 21:42:22.133130256 +0200
+++ rpm-5.4.10/configure.ac	2012-07-18 21:45:04.726457854 +0200
@@ -543,7 +543,7 @@
     MYPATH=$PATH
 fi
 
-DBXY=db61
+DBXY=db5.2
 
 AC_PATH_PROG(__BASH, bash, %{_bindir}/bash, $MYPATH)
 AC_PATH_PROG(__BZIP2, bzip2, %{_bindir}/bzip2, $MYPATH)
@@ -1582,7 +1582,7 @@
 CPPFLAGS="${CPPFLAGS} -I${prefix}/include/${DBXY}"
 RPM_CHECK_LIB(
     [Berkeley-DB], [db],
-    [db-6.1], [db_create], [db.h],
+    [db-5.2], [db_create], [db.h],
     [yes,external], [db3],
     [ DBLIBSRCS="$DBLIBSRCS db3.c"
       AM_CONDITIONAL(WITH_DB, [ true ])
@@ -1609,7 +1609,7 @@
 dnl XXX error: `db3' is already registered with AC_CONFIG_SUBDIRS.
 RPM_CHECK_LIB(
     [Berkeley-DB (+SQLite3)], [dbsql],
-    [db_sql-6.1], [sqlite3_open], [dbsql.h],
+    [db_sql-5.2], [sqlite3_open], [dbsql.h],
     [yes,external], [db3/sql],
     [
       AM_CONDITIONAL(WITH_DBSQL, [ true ])
