--- screen-4.0.2-orig/config.h.in	2004-01-19 16:27:08.106776000 -0500
+++ screen-4.0.2/config.h.in	2004-01-19 16:29:15.876778000 -0500
@@ -208,14 +208,16 @@
  * If screen is installed with permissions to update /etc/utmp (such
  * as if it is installed set-uid root), define UTMPOK.
  */
+#if 0 /* not sbit for screen per default */
 #define UTMPOK
+#endif
 
 /* Set LOGINDEFAULT to one (1)
  * if you want entries added to /etc/utmp by default, else set it to
  * zero (0).
  * LOGINDEFAULT will be one (1) whenever LOGOUTOK is undefined!
  */
-#define LOGINDEFAULT	1
+#define LOGINDEFAULT	0
 
 /* Set LOGOUTOK to one (1)
  * if you want the user to be able to log her/his windows out.
