diff -ur yum-3.2.23.orig/yum/packages.py yum-3.2.23/yum/packages.py
--- yum-3.2.23.orig/yum/packages.py	2009-05-14 18:41:21.000000000 +0200
+++ yum-3.2.23/yum/packages.py	2009-08-06 01:28:15.261213302 +0200
@@ -1200,12 +1200,13 @@
            is a pre-requires or a not"""
         # FIXME this should probably be put in rpmUtils.miscutils since 
         # - that's what it is
+        newflag = flag
         if flag is not None:
-            # Note: RPMSENSE_PREREQ == 0 since rpm-4.4'ish
-            if flag & (rpm.RPMSENSE_PREREQ |
-                       rpm.RPMSENSE_SCRIPT_PRE |
-                       rpm.RPMSENSE_SCRIPT_POST):
+            newflag = flag & rpm.RPMSENSE_PREREQ
+            if newflag == rpm.RPMSENSE_PREREQ:
                 return 1
+            else:
+                return 0
         return 0
 
     def _requires_with_pre(self):
