--- yum-3.4.3/rpmUtils/miscutils.py	2011-06-28 23:27:22.000000000 +0300
+++ yum-3.4.3/rpmUtils/miscutils.py	2013-04-27 22:33:20.000000000 +0300
@@ -377,6 +377,9 @@
 
     
 def flagToString(flags):
+    if flags & rpm.RPMSENSE_MISSINGOK:
+        return ('MISSINGOK', flagToString(flags & 0xf))
+
     flags = flags & 0xf
 
     if flags == 0: return None
--- yum-3.4.3/yum/packages.py~	2014-06-14 17:29:50.000000000 +0300
+++ yum-3.4.3/yum/packages.py	2014-06-14 17:34:50.475763987 +0300
@@ -1257,6 +1257,9 @@
                         continue
             prcostring = '''      <rpm:entry name="%s"''' % misc.to_xml(name, attrib=True)
             if flags:
+                # skip suggests
+                if flags[0] == 'MISSINGOK':
+                    continue
                 prcostring += ''' flags="%s"''' % misc.to_xml(flags, attrib=True)
                 if e:
                     prcostring += ''' epoch="%s"''' % misc.to_xml(e, attrib=True)
@@ -1464,14 +1464,18 @@
                 #  This is not the same as recommends/weak_requires.
                 bits = rpm.RPMSENSE_SCRIPT_PRE | rpm.RPMSENSE_SCRIPT_POST
                 weakreqs = [bool(flag & bits) for flag in lst]
+                suggests = [bool(flag & rpm.RPMSENSE_MISSINGOK) for flag in lst]
 
             vers = hdr[getattr(rpm, 'RPMTAG_%sVERSION' % tag)]
             prcotype = tag2prco[tag]
             self.prco[prcotype] = _end_nfv(name, lst, vers)
             if tag == 'REQUIRE':
                 weakreqs = zip(weakreqs, self.prco[prcotype])
+                suggests = zip(suggests, self.prco[prcotype])
                 strongreqs = [wreq[1] for wreq in weakreqs if not wreq[0]]
+                suggestreqs = [wreq[1] for wreq in suggests if not wreq[0]]
                 self.prco['strong_requires'] = strongreqs
+                self.prco['suggests'] = suggestreqs
 
         # This looks horrific as we are supporting both the old and new formats:
         tag2prco = { "SUGGEST":    ( misc.share_data("suggests"),
