diff -Nru eudev-3.1.1.orig/configure.ac eudev-3.1.1/configure.ac
--- eudev-3.1.1.orig/configure.ac	2015-06-08 13:46:49.535952811 +0200
+++ eudev-3.1.1/configure.ac	2015-06-08 13:47:03.698782701 +0200
@@ -244,10 +244,6 @@
 AM_CONDITIONAL([ENABLE_GUDEV], [test "x$enable_gudev" = "xyes"])
 
 # ------------------------------------------------------------------------------
-AC_PATH_TOOL(GPERF, gperf)
-if test -z "$GPERF" ; then
-        AC_MSG_ERROR([*** gperf not found])
-fi
 
 # ------------------------------------------------------------------------------
 GTK_DOC_CHECK([1.18],[--flavour no-tmpl])
diff -Nru eudev-3.1.1.orig/src/udev/Makefile.am eudev-3.1.1/src/udev/Makefile.am
--- eudev-3.1.1.orig/src/udev/Makefile.am	2015-06-08 13:46:49.532619596 +0200
+++ eudev-3.1.1/src/udev/Makefile.am	2015-06-08 13:47:03.698782701 +0200
@@ -112,18 +112,6 @@
 	keyboard-keys-from-name.h \
 	keyboard-keys-to-name.h
 
-keyboard-keys.txt: Makefile
-	$(AM_V_at)$(MKDIR_P) $(dir $@)
-	$(AM_V_GEN)$(CPP) $(CFLAGS) $(AM_CPPFLAGS) $(CPPFLAGS) -dM -include linux/input.h - < /dev/null | $(AWK) '/^#define[ \t]+KEY_[^ ]+[ \t]+[0-9]/ { if ($$2 != "KEY_MAX") { print $$2 } }' | sed 's/^KEY_COFFEE$$/KEY_SCREENLOCK/' > $@
-
-keyboard-keys-from-name.gperf: keyboard-keys.txt Makefile
-	$(AM_V_GEN)$(AWK) 'BEGIN{ print "struct key { const char* name; unsigned short id; };"; print "%null-strings"; print "%%";} { print tolower(substr($$1 ,5)) ", " $$1 }' < $< > $@
-
-keyboard-keys-from-name.h: keyboard-keys-from-name.gperf Makefile
-	$(AM_V_GPERF)$(GPERF) -L ANSI-C -t -N keyboard_lookup_key -H hash_key_name -p -C < $< > $@
-
-keyboard-keys-to-name.h: keyboard-keys.txt Makefile
-	$(AM_V_GEN)$(AWK) 'BEGIN{ print "const char* const key_names[KEY_CNT] = { "} { print "[" $$1 "] = \"" $$1 "\"," } END{print "};"}' < $< > $@
 
 
 BUILT_SOURCES = \
