--- dwm-5.8.2/Makefile~	2010-06-04 12:39:15.000000000 +0200
+++ dwm-5.8.2/Makefile	2010-12-27 10:54:57.840202001 +0100
@@ -10,13 +10,14 @@
 
 options:
 	@echo dwm build options:
-	@echo "CFLAGS   = ${CFLAGS}"
+	@echo "CFLAGS   = ${CFLAGS} ${OPTFLAGS}"
 	@echo "LDFLAGS  = ${LDFLAGS}"
+	@echo "LIBS	= ${LIBS}"
 	@echo "CC       = ${CC}"
 
 .c.o:
 	@echo CC $<
-	@${CC} -c ${CFLAGS} $<
+	@${CC} -c ${CFLAGS} ${OPTFLAGS} $<
 
 ${OBJ}: config.h config.mk
 
@@ -26,7 +27,7 @@
 
 dwm: ${OBJ}
 	@echo CC -o $@
-	@${CC} -o $@ ${OBJ} ${LDFLAGS}
+	@${CC} ${LDFLAGS} -o $@ ${OBJ} ${LIBS}
 
 clean:
 	@echo cleaning
--- dwm-5.9/config.mk~	2011-07-10 22:24:25.000000000 +0200
+++ dwm-5.9/config.mk	2011-08-12 13:49:51.859091895 +0200
@@ -21,9 +21,9 @@
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
 #CFLAGS = -g -std=c99 -pedantic -Wall -O0 ${INCS} ${CPPFLAGS}
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
+CFLAGS = -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
 #LDFLAGS = -g ${LIBS}
-LDFLAGS = -s ${LIBS}
+LDFLAGS = -s
 
 # Solaris
 #CFLAGS = -fast ${INCS} -DVERSION=\"${VERSION}\"
